pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/MessageDeduplication.java

```diff
@@ -337,12 +338,23 @@ public MessageDupStatus isDuplicate(PublishContext publishContext, ByteBuf heade
             publishContext.setOriginalHighestSequenceId(highestSequenceId);
             headersAndPayload.readerIndex(readerIndex);
         }
-
+        long chunkID = 0;
+        if (publishContext.isChunked()) {
+            if (md == null) {
+                headersAndPayload.markReaderIndex();
```

Comment:
Suggested use of this way to read the metadata to avoid the buffer being already marked once.

```java
int readerIndex = headersAndPayload.readerIndex();
....
headersAndPayload.readerIndex(readerIndex);
```